Style fixes for image grid and input field

GedalyaKrycer opened the pull request:
## Goal 

This looks to fix two styling issues that seemed to have appeared recently. 

#### Issues:

1. The browser's overflow scrollbars are impeding the image grid and causing it to lose one of its columns with a space on the right. (3 columns become 2 and 2 columns become 1). 
2. Package's input takes on the height if a default height applied to input[type="text"] was set by the parent site.


#### Proposed Solutions:
1. Refactor the image's container to use CSS Grid and remove associated styling from the images. 
2. Assign a default `height: inherent` to the search input and center its icons with flex's `align-items` property.

(More details are in the commit's descriptions.)


## Preview 
| Before                                                                                                           | After                                                                                                           |
|------------------------------------------------------------------------------------------------------------------|-----------------------------------------------------------------------------------------------------------------|
| ![Before](https://user-images.githubusercontent.com/29933355/139509333-fabd0a27-90a1-485e-8e97-3258fea0bd10.png) | ![After](https://user-images.githubusercontent.com/29933355/139509506-f4865547-914f-4143-a170-f8b2bcdb1032.png) |

GedalyaKrycer committed:
Control input height & center search icon

In some applications the input height can be overriden if a global style is set to `input[type="text"]`.  By adding `height: inherit` to the input's styles, we ensure the height doesn't unintentionly increase.

This commit also adds `display: flex` and `align-items: center` to the search and spinner icon's parent div. This ensures those will always be vertically centered, even in cases that the input height is increased. (Intentionally or otherwise)

GedalyaKrycer committed:
Refactor image container to use CSS Grids

For some reason the browsers scroll bar has started impeeding on the container that holds the grid of images. This was causing a 3 column grid to go to a 2 column grid, with a large empty space on the right.

By using CSS Grids, this simplifies the code needed to calculate how big the images should be and columns needed.

Instead of the `Photo` handling column width and gap per image, the parent container handles everything.

It sets the column sizes and dynamicly updates their width via a css variable that recieves the computed `searchResultWidth`.

This css variable sits in a `calc` to remove any column any space taken up by the browser's scroll bar.

End result is to ensure that the columns are preserved at all screen sizes and with browser scroll bars.

GedalyaKrycer committed:
Bump version: 0.2.0 -> 0.2.1

ligiapopescu commented:
Looking forward to this merge!

danielma commented on src/index.js:
```diff
@@ -527,37 +534,21 @@ Photo.propTypes = {
 }
 function Photo({
   photo,
-  width,
```
I'm happy to see the change to CSS grid here! I'm no expert, but it seems to solve a lot of common problems and the designers seem happy with it, so I'm on board 🙂 

At a high level, my feedback to this PR centers around the question: how much can we trust CSS grid to correctly size images for us? And I think this particular line is what sent me down that rabbit hole.

There's a lot of code in this file dedicated to sizing images properly within the grid. Now that we're setting width based on the grid, can we do the same thing with the height? I'd love to see us lean more heavily on CSS and get away from some of our custom sizing JS

GedalyaKrycer replied:
I think we can trust CSS Grid to do a good job sizing the width/columns accurately. However, in playing with this over the last few days, I ran into a lot of issues with sizing the rows/height. The issue seems to be that without an explicit height set on the image, it breaks out of its container and any CSS Grid Rows. 

However, we can use the same solution of passing the calculated `searchResultHeight` to a CSS variable and use it in a named class on the image. This can cut down on some JS in the form of props being used and clean up some styles that are directly applied to the image. 

The incoming commits include these proposed changes, along with some additional details.

GedalyaKrycer committed:
Remove unsused `width` prop

This was removed in a prevoius commit, in favor of passing the same value to css grid to create columns.

GedalyaKrycer committed:
Capture `searchResultHeight` in a css variable

This will allow us to use the calculated height for images in css classes, instead of prop chaining it to the image.

GedalyaKrycer committed:
Move image display/size properties to a css class

I tried to control the height with CSS Grid, but the image would keep breaking out of it's container, unless a height was set on the image explicitly.

Currently the image height is calculated based on the `photoRatio` and `searchResultWidth` values. This value is then passed down via props through the `<Photo>` component to the `<SpinnerImg>` component.

In the prevoius commit we saved this calculated height into a css variable. This lets us use it in a new named class called `.unsplash-react__image`.

To help with clarity within the file, I also move the display/size properties to a named class since they are all related.

I didn't move the border propeties to this class, since those are relying on JS values. It felt beyond the scope of this PR to do further restructuring.

GedalyaKrycer committed:
Remove no longer used `height` prop

dsecrest commented:
@GedalyaKrycer I have not looked at this on staging, so this is literally _just a shot in the dark_, but I'm curious about using `max-width` and `max-height` on the images. Would using those properties all the grid to control the images? If you tried that already, all good. Just a thought.

GedalyaKrycer commented:
> @GedalyaKrycer I have not looked at this on staging, so this is literally _just a shot in the dark_, but I'm curious about using `max-width` and `max-height` on the images. Would using those properties all the grid to control the images? If you tried that already, all good. Just a thought.

Yup, I did indeed try `max-height`/`max-width` on the image while using the calculated height passed to `grid-auto-rows` on the `unsplash-react__image-grid` container. Unfortunately, the images were still breaking out of their grid cells and retaining their original heights. 😞

GedalyaKrycer committed:
Remove unused columns & index props
